src/commands/dev/lib/builder-cache.ts

```diff
@@ -0,0 +1,150 @@
+import chalk from 'chalk';
+import execa from 'execa';
+import { join } from 'path';
+import npa from 'npm-package-arg';
+import mkdirp from 'mkdirp-promise';
+import { funCacheDir } from '@zeit/fun';
+import { readJSON, writeJSON, remove } from 'fs-extra';
+import cacheDirectory from 'cache-or-tmp-directory';
+import wait from '../../../util/output/wait';
+import { Output } from '../../../util/output';
+import { NowError } from '../../../util/now-error';
+import { devDependencies as nowCliDeps } from '../../../../package.json';
+import { Builder } from './types';
+
+import * as staticBuilder from './static-builder';
+
+const localBuilders: { [key: string]: Builder } = {
+  '@now/static': staticBuilder
+};
+
+export const cacheDirPromise = prepareCacheDir();
+export const builderDirPromise = prepareBuilderDir();
+
+/**
+ * Prepare cache directory for installing now-builders
+ */
+export async function prepareCacheDir() {
+  const designated = cacheDirectory('co.zeit.now');
+
+  if (!designated) {
+    throw new NowError({
+      code: 'NO_BUILDER_CACHE_DIR',
+      message: 'Could not find cache directory for now-builders.',
+      meta: {}
+    });
+  }
+
+  const cacheDir = join(designated, 'dev');
+  await mkdirp(cacheDir);
+  return cacheDir;
+}
+
+export async function prepareBuilderDir() {
+  const builderDir = join(await cacheDirPromise, 'builders');
+  await mkdirp(builderDir);
+
+  // Create an empty private `package.json`,
+  // but only if one does not already exist
+  try {
+    const buildersPkg = join(builderDir, 'package.json');
+    await writeJSON(buildersPkg, { private: true }, { flag: 'wx' });
+  } catch (err) {
+    if (err.code !== 'EEXIST') {
+      throw err;
+    }
+  }
+
+  return builderDir;
+}
+
+// Is responsible for cleaning the cache
+export async function cleanCacheDir(output: Output): Promise<void> {
+  try {
+    const cacheDir = await cacheDirPromise;
+    output.log(chalk`{magenta Deleting} ${cacheDir}`);
```

Comment:
```suggestion
    output.log(chalk.magenta(`Deleting ${cacheDir}`));
```

This looks like a typo to me 🤔

Reply:
It's not a typo, `chalk` supports this syntax. Also note that your proposal colors the cacheDir magenta, whereas the original code does not.